test/core/symbol/SymbolNetwork.spec.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 SYMBOL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { keccak256, sha3_256 } from 'js-sha3';
+import { SymbolNetworkList } from '../../../src/core/constants';
+import { SymbolNetwork } from '../../../src/core/symbol';
+import { BasicNetworkTester } from '../../BasicNetworkTest.template';
+
+describe('Symbol Network', () => {
+    const mainnet = SymbolNetworkList.find((network) => network.name === 'public') || SymbolNetworkList[0];
+    const testnet = SymbolNetworkList.find((network) => network.name === 'public_test') || SymbolNetworkList[1];
+
+    BasicNetworkTester(SymbolNetwork, mainnet, testnet);
+
+    describe('Address Hasher', () => {
+        it('can create correct hasher', () => {
+            // Arrange:
+            const network = new SymbolNetwork(mainnet.name, mainnet.identifier, mainnet.generationHash);
+            const expected = sha3_256.arrayBuffer(network.generationHash);
+            const unexpected = keccak256.arrayBuffer(network.generationHash);
+
+            // Act:
+            const hasher = network.addressHasher();
+            const hash = hasher.arrayBuffer(network.generationHash);
+
+            // Assert:
+            expect(hash).to.be.deep.equal(expected);
+            expect(hash).not.to.be.deep.equal(unexpected);
+        });
+    });
+
+    describe('Generation Hash', () => {
+        it('can find a generation hash by identifier', () => {
+            // Act:
+            const network = SymbolNetwork.findByIdentifier(mainnet.identifier);
+
+            // Assert:
+            expect(network?.generationHash).to.be.equal(mainnet.generationHash);
```

Comment:
probably better to check network is defined instead of `?.`

Reply:
agree, I will update it 👍🏼